app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -846,15 +929,41 @@ impl component::Frp<Model> for Frp {
             // === Selection ===
 
             selection_size_animation.target <+ groups.selection_size._1();
-            selection_animation.target <+ groups.selection_position_target.all_with(
+            selection_animation.target <+ groups.selection_position_target.all_with3(
                 &model.scroll_area.scroll_position_y,
-                f!([model]((id, pos), _) selection_position(&model,*id, *pos))
+                &layout_frp.update,
+                f!([model]((id, pos), _, style) selection_position(&model,*id, *pos, style))
             );
             selection_corners_animation.target <+ groups.selection_corners_radius._1();
-            eval selection_animation.value ((pos) selection.set_position_xy(*pos));
-            eval selection_size_animation.value ((pos) selection.size.set(*pos));
+            eval selection_animation.value ((pos) selection.pos.set(*pos));
+            eval selection_size_animation.value ((pos) selection.selection_size.set(*pos));
             eval selection_corners_animation.value ((r) selection.corners_radius.set(*r));
             eval_ model.scroll_area.scroll_position_y(selection_animation.skip.emit(()));
+            // The following code changes the `margin_top` parameter of the selection shape to
+            // handle the cases where the selection is near the top of the component list panel.
+            // See the documentation of the [`selection_box`](selection_box) module for more
+            // details.
```

Comment:
Every documentation should tell semantically what it does. It can redirect the reader to more detailed documentation, but it should at least shortly explain it. This doc does not explain things SEMANTICALLY. It tells what technically the follwoing code does, but without explanation of why. E.g. "handle the cases where the selection is near the top of the component list panel" - what cases? What is this even referring to?

Reply:
Rewritten